test/src/test/java/org/corfudb/integration/SecurityIT.java

```diff
@@ -0,0 +1,220 @@
+                .setKeyStore(serverPathToKeyStore)
+                .setKeyStorePassword(serverPathToKeyStorePassword)
+                .setTrustStore(serverPathToTrustStore)
+                .setTrustStorePassword(serverPathToTrustStorePassword)
+                .setLogPath(getCorfuServerLogPath(corfuSingleNodeHost, corfuStringNodePort))
+                .setSingle(true)
+                .runServer();
+    }
+
+    /**
+     * This is a helper method that load the properties required for creating corfu servers before
+     * each test by converting them from the values provided in CorfuDB.properties. Note that it
+     * will throw {@link NumberFormatException} if the numerical properties provided in CorfuDB.properties
+     * are not well-formed.
+     */
+    @Before
+    public void loadProperties() {
+        // Load host and port properties
+        corfuSingleNodeHost = PROPERTIES.getProperty("corfuSingleNodeHost");
+        corfuStringNodePort = Integer.parseInt(PROPERTIES.getProperty("corfuSingleNodePort"));
+
+        singleNodeEndpoint = String.format("%s:%d",
+                corfuSingleNodeHost,
+                corfuStringNodePort);
+
+        // Load TLS configuration, keystore and truststore properties
+        tlsEnabled = Boolean.valueOf(PROPERTIES.getProperty("tlsEnabledSecurityIT"));
+        serverPathToKeyStore = getPropertyAbsolutePath("serverPathToKeyStore");
+        serverPathToKeyStorePassword = getPropertyAbsolutePath("serverPathToKeyStorePassword");
+        serverPathToTrustStore = getPropertyAbsolutePath("serverPathToTrustStore");
+        serverPathToTrustStorePassword = getPropertyAbsolutePath("serverPathToTrustStorePassword");
+        runtimePathToKeyStore = getPropertyAbsolutePath("runtimePathToKeyStore");
+        runtimePathToKeyStorePass = getPropertyAbsolutePath("runtimePathToKeyStorePass");
+        runtimePathToTrustStore = getPropertyAbsolutePath("runtimePathToTrustStore");
+        runtimePathToTrustStorePass = getPropertyAbsolutePath("runtimePathToTrustStorePass");
+    }
+
+    // Take the property string provided int CorfuDB.properties and return the absolute path string
+    private String getPropertyAbsolutePath(String pathProperty) {
+        return Paths.get(PROPERTIES.getProperty(pathProperty)).toAbsolutePath().toString();
+    }
+
+    /**
+     * This test creates Corfu runtime and a single Corfu server according to the configuration
+     * provided in CorfuDB.properties. Corfu runtime configures TLS related parameters using
+     * {@link CorfuRuntime}'s API and then asserts that operations on a CorfuTable is executed
+     * as Expected.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testServerRuntimeTlsEnabledMethod() throws Exception {
+        // Run a corfu server
+        Process corfuServer = runSinglePersistentServerTls();
+
+        // Start a Corfu runtime
+        CorfuRuntime corfuRuntime = new CorfuRuntime(singleNodeEndpoint)
```

Comment:
Create a method to return runtime with TLS enabled.

Reply:
In testServerRuntimeTlsEnabledMethod vs testServerRuntimeTlsEnabledByParameter, I was testing (and also showing how we can use both approaches) for enabling tls. For that, it will make a more coherent flow (and better example) to keep the creation of runtime within each test case (I have done the same for later test using parameters). That being said, if you feel strongly about it, I will take runtime creation of both into separate methods. let me know.